src/main/java/co/cask/wrangler/steps/Mask.java

```diff
@@ -0,0 +1,164 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.wrangler.steps;
+
+import co.cask.wrangler.api.AbstractStep;
+import co.cask.wrangler.api.Row;
+import co.cask.wrangler.api.SkipRowException;
+import co.cask.wrangler.api.StepException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Random;
+
+/**
+ * A Wrangler plugin that applies substitution and shuffling masking on the column.
+ *
+ * <p>
+ *  Substitution masking us generally used for masking credit card or SSN numbers.
+ *  This type of masking is fixed masking, where the pattern is applied on the
+ *  fixed length string.
+ *
+ *  <ul>
+ *    <li>Use of # will include the digit from the position.</li>
+ *    <li>Use x to mask the digit at that position</li>
+ *    <li>Any other characters will be inserted as-is.</li>
+ *  </ul>
+ *
+ *  <blockquote>
+ *    <pre>
+ *        Step step = new Mask(lineno, line, "ssn", "XXX-XX-####", 1);
+ *        Step step = new Mask(lineno, line, "amex", "XXXX-XXXXXX-X####", 1);
+ *    </pre>
+ *  </blockquote>
+ * </p>
+ *
+ * <p>
+ *   Fixed length shuffle masking performs obfuscation by using random character
+ *   substitution method. The data is randomly shuffled in the column.
+ *
+ *   <blockquote>
+ *     <pre>
+ *       Step step = new Mask(lineno, line, "150 Mars Avenue, Marcity, Mares", 2);
+ *     </pre>
+ *   </blockquote>
+ * </p>
+ */
+public class Mask extends AbstractStep {
+  private static final Logger LOG = LoggerFactory.getLogger(Mask.class);
+
+  // Specifies types of mask
+  public static final int MASK_NUMBER = 1;
+  public static final int MASK_SHUFFLE = 2;
+
+  // Masking pattern
+  private final String mask;
+
+  // Column on which to apply mask.
+  private final String column;
+
+  // Type of mask.
+  private final int maskType;
+
+  public Mask(int lineno, String detail, String column, int maskType) {
+    super(lineno, detail);
+    this.mask = "";
+    this.column = column;
+    this.maskType = maskType;
+  }
+
+  public Mask(int lineno, String detail, String column, String mask, int maskType) {
+    super(lineno, detail);
+    this.mask = mask;
+    this.column = column;
+    this.maskType = maskType;
+  }
+
+  /**
+   * Masks the column specified using either substitution method or shuffling.
+   *
+   * @param row Input {@link Row} to be wrangled by this step.
+   * @return A newly transformed {@link Row} with masked column.
+   * @throws StepException thrown when there is issue with masking
+   * @throws SkipRowException thrown when the row needs to be skipped
```

Comment:
It's a contract, if it's not thrown then it's fine. There are other steps that might use this to skip rows.